src/main/java/bg/codexio/springframework/data/jpa/requery/adapter/GraphQLHttpFilterAdapter.java

```diff
@@ -0,0 +1,324 @@
+    private <T> FilterRequestWrapper<T> processPostRequest(HttpServletRequest request) {
+        try {
+            var requestBody = new StringBuilder();
+            var reader = request.getReader();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                requestBody.append(line);
+            }
+
+            var jsonMap = this.objectMapper.readValue(
+                    requestBody.toString(),
+                    new TypeReference<Map<String, Object>>() {}
+            );
+
+            var query = (String) jsonMap.get("query");
+            var extractedFilter = extractFilterBody(query);
+            if (extractedFilter.isPresent()) {
+                return this.graphQLComplexFilterAdapter.adapt(extractedFilter.get());
+            }
+
+            return new FilterRequestWrapper<>(parseGraphQLQuery(query));
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+
+            return new FilterRequestWrapper<>();
+        }
+    }
+
+    private List<FilterRequest> parseGraphQLQuery(String query) {
+        var parser = new Parser();
+        var document = parser.parseDocument(query);
+
+        return processDocument(document);
+    }
+
+    private List<FilterRequest> processDocument(Document document) {
+        var filterRequests = new ArrayList<FilterRequest>();
+        var operationDefinitionList =
+                document.getDefinitionsOfType(OperationDefinition.class);
+        for (var operation : operationDefinitionList) {
+            if (!operation.getOperation()
+                          .equals(OperationDefinition.Operation.QUERY)) {
+                continue;
+            }
+            for (var field : operation.getSelectionSet()
+                                      .getSelectionsOfType(Field.class)) {
+                for (var argument : field.getArguments()) {
+                    var value = argument.getValue();
+                    var extractedValue = extractValue(value);
+                    if (extractedValue instanceof Map) {
+                        handleMapValue(
+                                argument.getName(),
+                                extractedValue,
+                                filterRequests
+                        );
+                        continue;
+                    }
+                    var filterRequest = new FilterRequest(
+                            extractName(argument.getName()),
+                            extractedValue,
+                            getOperationFromArgument(argument.getName())
+                    );
+                    filterRequests.add(filterRequest);
+                }
+            }
+        }
+        return filterRequests;
+    }
+
+    private void handleMapValue(
+            String containingObjectName,
+            Object extractedValue,
+            List<FilterRequest> filterRequests
+    ) {
+        if (extractedValue instanceof Map) {
+            for (var entry :
```

Comment:
I personally don't like the whole code execution to be indented because of a single conditional. Better short-circuit with return.

Reply:
Also I believe you can use a smart cast?

Reply:
used short-circuit, unable to use smart cast because of the generics in the map - done